core/src/main/java/com/novoda/landingstrip/Notifier.java

```diff
@@ -0,0 +1,34 @@
+package com.novoda.landingstrip;
+
+import android.view.View;
+
+class Notifier implements LandingStripAdapter.Listener {
+
+    private final TabsContainerView tabsContainerView;
+
+    Notifier(TabsContainerView tabsContainerView) {
+        this.tabsContainerView = tabsContainerView;
+    }
+
+    @Override
+    public <T extends View> void onNotifyDataSetChanged(LandingStripAdapter<T> adapter) {
+        recreateAndBindTabs(adapter);
+    }
+
+    private <T extends View> void recreateAndBindTabs(LandingStripAdapter<T> adapter) {
```

Comment:
I totally agree but think the recycling effort would be more appreciated here https://github.com/novoda/view-pager-adapter